### PR TITLE
Support the creation of top-level folders on Windows (fixes #1822)

### LINKED ISSUE #1822: Creating a new folder on C:\ fails (Windows 7)
I added a new folder with the following path: "C:\test"
Syncthing fails to create this folder with the following output: 
```INFO: Skipping folder dsad scan due to folder error: mkdir \\?: Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch. ```

Which means: wrong syntax for filename, directory or volume label.

Creating subfolders works.

### PR DESCRIPTION
**Pull request by Zillode**

### CODE REVIEW

File: internal/osutil/mkdirall.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package osutil
```
Comment (calmh) [inline internal/osutil/mkdirall.go]: Needs a `// +build !windows`

File: internal/osutil/mkdirall_windows.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+package osutil
+
+import (
+	"os"
+	"syscall"
+)
+
+// See https://github.com/golang/go/issues/10900
+
+// MkdirAll creates a directory named path,
```
Comment (calmh) [inline internal/osutil/mkdirall_windows.go]: Freaky wrapping here. 80 columns is usual. :)

File: internal/osutil/mkdirall_windows.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	j := i
+	for j > 0 && !os.IsPathSeparator(path[j-1]) { // Scan backward over element.
+		j--
+	}
+
+	if j > 1 {
+		// Create parent
+		parent := path[0 : j-1]
+		if parent != os.VolumeName(parent) {
+			err = MkdirAll(parent, perm)
```
Comment (calmh) [inline internal/osutil/mkdirall_windows.go]: I'm not super fond of the recursiveness here... How about some like ... (pseudocode)

```go
cleaned := filepath.Clean(path)
if strings.HasPrefix(cleaned, `\\?\`) { cleaned = cleaned[4:] }
split := strings.Split(cleaned, os.PathSeparator)
for i := range split {
  path := `\\?\` + strings.Join(split[:i], os.PathSeparator)
  if os.Stat(path) is happy { continue }
  err := os.Mkdir(path)
 ...
}
```
?

Comment (Zillode) [inline internal/osutil/mkdirall_windows.go]: This is based on https://github.com/golang/go/blob/91191e7b7bc8c0e1a6d49c7a9b3adeb1ab39a423/src/os/path.go#L19
`if parent != os.VolumeName(parent) {` is the added check to fix the bug

Comment (calmh) [inline internal/osutil/mkdirall_windows.go]: Oh. Interesting. Well in that case I guess it's fine, as the minimal modification is better.

**Comment by calmh** [comment]
This has some build errors.

**Comment by calmh** [comment]
:+1:

**Pull request merged**